lib/trento/discovery/policies/cluster_policy.ex

```diff
@@ -450,16 +452,34 @@ defmodule Trento.Discovery.Policies.ClusterPolicy do
     |> Enum.find("Unknown", fn site -> site != primary_site end)
   end
 
-  defp parse_cluster_fencing_type(%{resources: resources}) do
-    Enum.find_value(resources, "", fn
+  defp parse_cluster_fencing_type(%{resources: resources}, sbd) do
+    resources
+    |> Enum.find_value(nil, fn
       %{agent: "stonith:" <> fencing_type} ->
         fencing_type
 
       _ ->
         false
     end)
+    |> do_fencing_type(sbd)
+  end
+
+  defp do_fencing_type(nil, sbd) do
+    if diskless_sbd?(sbd) do
+      @diskless_sbd_fencing_type
+    else
+      @unknown_fencing_type
+    end
   end
 
+  defp do_fencing_type(fencing_type, _), do: fencing_type
+
+  defp diskless_sbd?(%{config: %{"sbd_watchdog_dev" => _} = config, devices: []}) do
+    not Map.has_key?(config, "sbd_device")
```

Comment:
Could we write that condition with pattern matching?

Like config has the key `sbd_device` and the other clause without the key

Reply:
We could also has one function that extracts the fencing type and retuurn diskless fencing type or unknown, we can skip the if at line 469

Reply:
done. initially i didn't want to put the `unknown` case first, as i wanted to put the positive cases first, but yes, eventually the code is now a bit cleaner